src/ui/Analytics/LiveAnalyticsClient.ts

```diff
@@ -151,11 +152,17 @@ export class LiveAnalyticsClient implements IAnalyticsClient {
     var customEvent = this.buildCustomEvent(actionCause, metaObject, element);
     this.triggerChangeAnalyticsCustomData('CustomEvent', metaObject, customEvent);
     this.checkToSendAnyPendingSearchAsYouType(actionCause);
-    Defer.defer(() => {
-      if (this.sendToCloud) {
-        this.endpoint.sendCustomEvent(customEvent);
-      }
-      $$(this.rootElement).trigger(AnalyticsEvents.customEvent, <IAnalyticsCustomEventArgs>{ customEvent: APIAnalyticsBuilder.convertCustomEventToAPI(customEvent) });
+    return new Promise((resolve) => {
+      Defer.defer(() => {
```

Comment:
What was the point the of "Defer" here ? Looks like a fancy `setTimeout(..., 0)` to me.
Now that we are executed inside a Promise, doesn't it do the same thing ?
I might be wrong but could we remove the Defer call now ?